partcad/src/partcad/project.py

```
@@ -1427,15 +1273,31 @@
 
-    def test(self):
+    async def _run_test_async(self, ctx, tests: list, use_wrapper: bool = False) -> list[asyncio.Task]:
+        tasks = []
+        test_method = "test_log_wrapper" if use_wrapper else "test"
         for interface in self.interfaces.values():
-            interface.test()
+            tasks.append(asyncio.create_task(interface.test_async()))
         for sketch in self.sketches.values():
-            sketch.test()
+            tasks.extend(asyncio.create_task(getattr(t, test_method)(tests, ctx, sketch)) for t in tests)
         for part in self.parts.values():
```

Ruff (0.8.2)

Import `part` from line 33 shadowed by loop variable

(F402)